mr/src/test/java/org/apache/iceberg/mr/hive/TestHiveIcebergOutputFormat.java

```diff
@@ -0,0 +1,285 @@
+import org.apache.iceberg.mr.hive.HiveIcebergOutputFormat.IcebergRecordWriter;
+import org.apache.iceberg.mr.mapreduce.IcebergWritable;
+import org.apache.iceberg.relocated.com.google.common.collect.Sets;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+@RunWith(Parameterized.class)
+public class TestHiveIcebergOutputFormat {
+
+  private static final Object[] TESTED_FILE_FORMATS = new Object[] {"avro", "orc", "parquet"};
+
+  @Rule
+  public TemporaryFolder temp = new TemporaryFolder();
+
+  @Parameterized.Parameters(name = "{0}")
+  public static Object[] parameters() {
+    return TESTED_FILE_FORMATS;
+  }
+
+  // parametrized variables
+  private final FileFormat fileFormat;
+
+  private static final Configuration conf = new Configuration();
+  private TestHelper helper;
+  private Table table;
+  private TestOutputFormat testOutputFormat;
+
+  public TestHiveIcebergOutputFormat(String fileFormat) {
+    this.fileFormat = FileFormat.valueOf(fileFormat.toUpperCase(Locale.ENGLISH));
+  }
+
+  @Before
+  public void before() throws Exception {
+    helper = new TestHelper(conf,
+        new HadoopTables(conf),
+        temp.newFolder(this.fileFormat.name()).toString(),
+        HiveIcebergSerDeTestUtils.FULL_SCHEMA,
+        null,
+        fileFormat,
+        temp);
+
+    table = helper.createUnpartitionedTable();
+    testOutputFormat = new TestOutputFormat(table, fileFormat);
+  }
+
+  @Test
+  public void testWriteRow() throws IOException {
+    // Write a row.
+    List<Record> records =
+        Arrays.asList(new Record[] { HiveIcebergSerDeTestUtils.getTestRecord(FileFormat.PARQUET.equals(fileFormat)) });
+
+    testOutputFormat.write(records, false, false);
+    testOutputFormat.validate(records);
+  }
+
+  @Test
+  public void testNullRow() throws IOException {
+    // FIXME: ORC file does not read back the row consisting only of nulls. The data in the files seems ok.
+    if (FileFormat.ORC.equals(fileFormat)) {
+      return;
+    }
+    // Write a row.
+    List<Record> records = Arrays.asList(new Record[] { HiveIcebergSerDeTestUtils.getNullTestRecord() });
+
+    testOutputFormat.write(records, false, false);
+    testOutputFormat.validate(records);
+  }
+
+  @Test
+  public void testMultipleRows() throws IOException {
+    // Write 2 rows. One with nulls too.
+    List<Record> records = Arrays.asList(new Record[] {
+        HiveIcebergSerDeTestUtils.getTestRecord(FileFormat.PARQUET.equals(fileFormat)),
+        HiveIcebergSerDeTestUtils.getNullTestRecord()
+    });
+
+    testOutputFormat.write(records, false, false);
+    testOutputFormat.validate(records);
+  }
+
+  @Test
+  public void testRandomRecords() throws IOException {
+    // Write 30 random rows
+    // FIXME: Parquet appender expect byte[] instead of UUID when writing values.
```

Comment:
Is this something you still want to fix in this PR or in another one?

Reply:
I am not even sure that this could be changed without broader community approval - might be a breaking change for existing parquet writers